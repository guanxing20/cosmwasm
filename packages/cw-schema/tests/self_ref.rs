#![allow(dead_code)]

#[derive(cw_schema::Schemaifier)]
pub struct SelfRef {
    meow: Vec<SelfRef>,
}

#[test]
fn selfref() {
    insta::assert_json_snapshot!(cw_schema::schema_of::<SelfRef>());
}
