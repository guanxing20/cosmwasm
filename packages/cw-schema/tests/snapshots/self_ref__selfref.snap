---
source: packages/cw-schema/tests/self_ref.rs
expression: "cw_schema::schema_of::<SelfRef>()"
---
{
  "type": "v1",
  "root": 0,
  "definitions": [
    {
      "name": "self_ref_SelfRef",
      "type": "struct",
      "properties": {
        "meow": {
          "value": 1
        }
      }
    },
    {
      "name": "alloc::vec::Vec<self_ref::SelfRef>",
      "type": "array",
      "items": 0
    }
  ]
}
